### PR TITLE
fix(jsmn-find): fix RECALLOC_OR_ERROR

### PR DESCRIPTION
**Pull request by Ratakor**
## Notice
- [x] I *understand* the code that I have edited, and have the means
to test it before making changes to Concord.

## What?
The code is clearer than an explanation.

## Anything Else
Size for memset could be `*(prev_size) * sizeof *(ptr)` instead of `(new_size - *(prev_size)) * sizeof *(ptr)` but it's not that important.

### CODE REVIEW

File: core/jsmn-find.h [hunk 1]
```diff
@@ -405,14 +405,12 @@ jsmnf_find_path(const struct jsmnf_pair *head,
...
 #define RECALLOC_OR_ERROR(ptr, prev_size)                                     \
     do {                                                                      \
-        const unsigned new_size = *prev_size * 2;                             \
-        void *tmp = realloc((ptr), new_size * sizeof *(ptr));                 \
-        if (!tmp) return JSMN_ERROR_NOMEM;                                    \
-                                                                              \
-        *prev_size = new_size;                                                \
+        const unsigned new_size = *(prev_size) * 2;                           \
+        (ptr) = realloc((ptr), new_size * sizeof *(ptr));                     \
+        if (!(ptr)) return JSMN_ERROR_NOMEM;                                  \
```
[the remark this answers is not in the document]
Comment (Ratakor) [reply core/jsmn-find.h]: Ah yea, fixed it.

**Review by lcsmuller — APPROVED** [review]

**Comment by lcsmuller** [comment]
Looks good to me, thank you! If it's not asking much could you also add this patch to [the original repo](https://github.com/lcsmuller/jsmn-find)?

**Comment by Ratakor** [comment]
> Looks good to me, thank you! If it's not asking much could you also add this patch to [the original repo](https://github.com/lcsmuller/jsmn-find)?

Sure, done.

**Comment by lcsmuller** [comment]
Please sync your fork with latest `dev`!
````console
git fetch upstream
git checkout dev
git rebase upstream/dev
```

**Comment by Ratakor** [comment]
Github is so awkward I think it's good now.

**Pull request merged**